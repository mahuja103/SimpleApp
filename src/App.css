.App{
     text-align: center;
     padding-top: 20%;
 }
.App button{
    padding: 10px 20px;
    border-style: hidden;
    color: white;
    background: blue;
    cursor: pointer;
}