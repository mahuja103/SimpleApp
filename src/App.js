import React from 'react';
import './App.css';

class App extends React.Component {
  state={
    hide:"show",
      show:false
  }
  show(){
    switch (this.state.show){
        case true:
            this.setState({show: false, hide:"show"})
           break;
        case false:
            this.setState({show: true, hide:"hide"})
            break;
    }

    }
    render(){
        return (
            <div className="App">
                <button onClick={this.show.bind(this)}>Click Me to {this.state.hide}</button>
                {
                    this.state.show ?      <h2>Hello CircleCI!</h2>: null
                }

            </div>
        );

    }

}

export default App;
