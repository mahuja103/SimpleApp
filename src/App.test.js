import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';


test("testing the text shown or not",()=>{
  const tree= renderer.create(<App/>).toJSON();
  const checker= tree.children;
  console.log( checker);
    expect(checker[0].type).toEqual('button');


})
test("testing button ",()=>{
  const tree= renderer.create(<App/>).toJSON();
  const checker= tree.children;
  console.log(checker);
     expect(checker[0].children[1]).toEqual('show');


})
