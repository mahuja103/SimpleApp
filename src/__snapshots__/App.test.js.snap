// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the text shown or not 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Click Me to  
    show
  </button>
</div>
`;
